### PR TITLE
Issue #PS-1093 feat: Implement list view for centers in selected block within my district

### PR DESCRIPTION
**Pull request by Aar-if**
Implement list view for centers in selected block within my district

## Summary by CodeRabbit

- **New Features**
  - Introduced search functionality and center type filtering.
  - Added internationalization support for search terms and center types in English, Hindi, Marathi, and Oriya.

- **Bug Fixes**
  - Updated placeholder text for search input field.

- **Style**
  - Added new color values to the primary theme palette.
  
- **New Components**
  - Created `FilterModalCenter` for filtering centers.
  - Introduced `Blocks` component for managing and displaying centers.

### CODE REVIEW

File: src/pages/blocks/centre.tsx [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+  const theme = useTheme<any>();
+  const [districts, setDistrict] = useState('');
+  const [blocks, setBlocks] = useState('');
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null);
+  const [filterModalOpen, setFilterModalOpen] = useState(false); // State for filter modal
+  const [searchInput, setSearchInput] = useState(''); // State for search input
+  const [centers, setCenters] = useState([
+    'Bhivapur',
+    'Kamptee',
+    'Katol',
```
Comment (itsvick) [inline src/pages/blocks/centre.tsx:33]: @Aar-if this should be array of objects

File: src/pages/blocks/centre.tsx [hunk 2]
```diff
@@ -0,0 +1,327 @@
...
+    );
+  };
+
+  const applyFilters = () => {
+    // Apply sorting and filtering based on selected options
+    let updatedCenters = centers;
+
+    // Apply sorting
+    if (sortOrder === 'asc') {
+      updatedCenters = updatedCenters?.sort();
```
Comment (itsvick) [inline src/pages/blocks/centre.tsx:68]: @Aar-if when centers will be array of objects it wont work, once u change the centers array to array of object, change this logic accordingly

File: src/pages/blocks/centre.tsx [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+    } else if (sortOrder === 'desc') {
+      updatedCenters = updatedCenters?.sort().reverse();
+    }
+
+    // Apply center type filter
+    if (centerType === 'regular') {
+      updatedCenters = updatedCenters?.filter((center) =>
+        ['Bhivapur', 'Kamptee', 'Katol'].includes(center)
```
Comment (itsvick) [inline src/pages/blocks/centre.tsx:76]: @Aar-if dont hard code the center, instead add new key in the center object as `remote: true` and use same here

File: src/pages/blocks/centre.tsx [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+            {blocks}
+          </Typography>
+          <Typography
+            textAlign={'left'}
+            fontSize={'11px'}
+            mb={'1rem'}
+            fontWeight={'500'}
+            color={theme?.palette?.text?.primary}
+          >
+            {districts}, Maharashtra
```
Comment (itsvick) [inline src/pages/blocks/centre.tsx:129]: @Aar-if avoid hard coding here

File: src/pages/blocks/centre.tsx [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+          placeholder={t('COMMON.SEARCH')}
+          variant="outlined"
+          size="medium"
+          sx={{
+            p: 2,
+                justifyContent: 'center',
+                height: '48px',
+                flexGrow: 1,
+                mr: 1,
+                backgroundColor: '#EDEDED',
```
Comment (itsvick) [inline src/pages/blocks/centre.tsx:144]: @Aar-if color should be in the custom theme

File: src/pages/blocks/centre.tsx [hunk 6]
```diff
@@ -0,0 +1,327 @@
...
+        />
+          <Button
+              variant="outlined"
+              onClick={handleFilterModalOpen}
+              size="medium"
+              endIcon={<ArrowDropDown />}
+              sx={{
+                backgroundColor: 'white',
+                borderRadius: '7px',
+                border: '1px solid #EDEDED',
```
Comment (itsvick) [inline src/pages/blocks/centre.tsx:169]: @Aar-if colors to be moved to custom theme

File: src/pages/blocks/centre.tsx (hunk 1 above)
Comment (Aar-if) [reply src/pages/blocks/centre.tsx:33]: Okay

File: src/pages/blocks/centre.tsx (hunk 2 above)
Comment (Aar-if) [reply src/pages/blocks/centre.tsx:68]: Okay

File: src/pages/blocks/centre.tsx (hunk 3 above)
Comment (Aar-if) [reply src/pages/blocks/centre.tsx:76]: Yes

File: src/pages/blocks/centre.tsx [hunk 7]
```diff
@@ -0,0 +1,327 @@
...
+                  mr: 1,
+                }}
+              >
+                <SmartDisplayOutlined fontSize="small" /> {/* Left icon */}
+              </Box>
+              {center}
+            </Box>
+            <KeyboardArrowRight /> {/* Right icon */}
+          </Button>
+        ))}
```
[the remark this answers is not in the document]
Comment (itsvick) [reply src/pages/blocks/centre.tsx:297]: @Aar-if check code-rabbit

File: src/pages/blocks/centre.tsx (hunk 4 above)
Comment (Aar-if) [reply src/pages/blocks/centre.tsx:129]: Yes

File: src/pages/blocks/centre.tsx (hunk 5 above)
Comment (Aar-if) [reply src/pages/blocks/centre.tsx:144]: Okay I'll check

File: src/pages/blocks/centre.tsx (hunk 6 above)
Comment (Aar-if) [reply src/pages/blocks/centre.tsx:169]: Okay

File: src/pages/blocks/centre.tsx [hunk 8]
```diff
@@ -0,0 +1,338 @@
...
+interface BlocksProps {
+  // Define any props if needed
+}
+
+interface Center {
+  name: string;
+  type: CenterType;
+}
+
+type CenterType = '' | 'regular' | 'remote';
```
Comment (itsvick) [inline src/pages/blocks/centre.tsx:32]: @Aar-if move this to app.constant file with enum

**Pull request merged**